### PR TITLE
Security validation fixes

### PR DESCRIPTION
**Pull request by grrowe**
- https://aptible.zendesk.com/agent/tickets/50991
- https://aptible.zendesk.com/agent/tickets/50992

- Added validation to forgot-pass, signup, and edit name
- Added sanitize function to remove harmful characters
- Modified invalid_email response from forgot-pass to show a `fakeTrue` so a potentially harmful user cannot know whether an email is a user.
    - they can still check the network tab but a change in `auth-api` will still be needed

### CODE REVIEW

**Review by joeclayallday — APPROVED** [review]

File: src/ui/pages/forgot-pass.tsx [hunk 1]
```diff
@@ -45,7 +46,7 @@ export const ForgotPassPage = () => {
                 type="email"
                 name="email"
                 value={email}
-                onChange={(e) => setEmail(e.currentTarget.value)}
+                onChange={(e) => setEmail(sanitizeInput(e.currentTarget.value))}
```
Comment (neurosnap) [inline src/ui/pages/forgot-pass.tsx:49]: Let's get a test written for this change.

File: src/ui/pages/settings-profile.tsx [hunk 2]
```diff
@@ -22,14 +24,26 @@ export function SettingsProfilePage() {
...
+      existValidtor(props.name, "Name") || nameValidator(props.name),
+  };
+  const [errors, validate] = useValidator<UpdateNameForm, typeof validators>(
+    validators,
+  );
+  const isDisabled = name === user.name || !!errors.name;
+
+  interface UpdateNameForm {
+    name: string;
+  }
```
Comment (neurosnap) [inline src/ui/pages/settings-profile.tsx:38]: Let's add a test for this change.

File: src/ui/pages/signup.tsx
Comment (neurosnap) [inline src/ui/pages/signup.tsx:1]: Let's write a test for this change

**Review by neurosnap — CHANGES_REQUESTED** [review]
Thanks so much for these fixes!  Let's write some automated tests for them, happy to help pair or collaborate on writing them.

File: src/ui/pages/forgot-pass.test.tsx [hunk 3]
```diff
@@ -28,13 +28,33 @@ describe("ForgotPassPage", () => {
...
-      /Success! Check your email to change your password./,
+      /If an Aptible account exists for that email address, we will email you instructions for resetting your password./,
     );
     expect(
       btn,
       "when already reset password the btn should be disabled",
     ).toBeDisabled();
   });
+
+  it("should hide a User not found email", async () => {
```
Comment (neurosnap) [inline src/ui/pages/forgot-pass.test.tsx:39]: This one we actually need to create a stub response that includes the `invalid_email` error:

```ts
server.use(
  rest.post(`${testEnv.authUrl}/password/resets/new`, (_, res, ctx) => {
    return res(ctx.json({ error: "invalid_email" }));
  }),
);
```

File: src/ui/pages/settings-profile.test.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { act, fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { setupIntegrationTest } from "@app/test";
+import { SettingsProfilePage } from "./settings-profile";
+
+describe("ForgotPassPage", () => {
```
Comment (neurosnap) [inline src/ui/pages/settings-profile.test.tsx:7]: Name here doesn't match the test file

**Review by neurosnap — CHANGES_REQUESTED** [review]
Looks pretty good!  I have 2 follow up changes

File: src/auth/pass.ts [hunk 5]
```diff
@@ -22,15 +21,24 @@ export const forgotPass = authApi.post<{ email: string }>(
...
-      ctx.loader = {
-        message: `Error! Unable to submit request to reset your password: ${ctx.json.error.message}
-        `,
-      };
+      if (ctx.json.error.error === "invalid_email") {
+        ctx.loader = {
+          message:
+            "If an Aptible account exists for that email address, we will email you instructions for resetting your password.",
+          meta: { fakeSuccess: true },
+        };
```
Comment (neurosnap) [inline src/auth/pass.ts:29]: While testing, make sure this code path is getting hit

**Review by neurosnap — APPROVED** [review]

**Pull request merged**